crates/cairo-lint-core/src/lints/double_parens.rs

```diff
@@ -0,0 +1,28 @@
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_syntax::node::ast::Expr;
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use cairo_lang_syntax::node::{TypedStablePtr, TypedSyntaxNode};
+
+pub const DOUBLE_PARENS: &str = "unnecessary double parentheses found. Consider removing them.";
+
+pub fn check_double_parens(db: &dyn SyntaxGroup, expr: &Expr, diagnostics: &mut Vec<PluginDiagnostic>) {
+    let span = match expr {
+        Expr::Parenthesized(parenthesized_expr) => {
+            if let Expr::Parenthesized(_) | Expr::Tuple(_) = parenthesized_expr.expr(db) {
+                Some(parenthesized_expr.as_syntax_node().span(db))
+            } else {
+                None
+            }
+        }
+        _ => None,
+    };
+
+    if let Some(_span) = span {
+        diagnostics.push(PluginDiagnostic {
+            stable_ptr: expr.stable_ptr().untyped(),
+            message: DOUBLE_PARENS.to_string(),
+            severity: Severity::Warning,
+        });
```

Comment:
I think this should be refactored to use a boolean instead of a useless `Option`. Also the `match + if let` can be combined in 1 `if let`